fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java

```diff
@@ -650,4 +657,36 @@ private static List<BucketMetadata> getBucketMetadataFromContext(
                 });
         return bucketMetadataList;
     }
+
+    /**
+     * Validates whether the table creation is allowed based on the table type and configuration.
+     *
+     * @param tableDescriptor the table descriptor to validate
+     * @param tablePath the table path for error reporting
+     * @throws InvalidTableException if table creation is not allowed
+     */
+    private void validateTableCreationPermission(
+            TableDescriptor tableDescriptor, TablePath tablePath) {
+        boolean hasPrimaryKey = tableDescriptor.hasPrimaryKey();
+
+        if (hasPrimaryKey) {
+            // This is a KV table (Primary Key Table)
+            if (!kvTableAllowCreation) {
+                throw new InvalidTableException(
+                        String.format(
+                                "Creation of KV tables (primary key tables) is not allowed. "
+                                        + "Table: %s. Please set '%s' to true to enable KV table creation.",
+                                tablePath, ConfigOptions.KV_TABLE_ALLOW_CREATION.key()));
```

Comment:
```suggestion
                        "Creation of Primary Key Tables is disallowed in the cluster.");
```

1. Use `Primary Key Table` for user-facing messages, we don't have the concept of KV table actually. 
2. remove the cluster config from the exception message, because user shouldn't be aware of the cluster config and usually don't have the permission to access cluster configs (so it's not necessary to expose them).